### DISCUSSION TITLE
A simple algorithm for distances and worldlines in FLRW

### DISCUSSION THREAD

**Discussion by mikehelland** [a-cosmology-group/acg#155] [Cosmology Models]
Edit: I've updated the algorithm to work for open and closed FLRW models too. It's still just as simple.

Python notebook:
https://github.com/mikehelland/hubbles-law/blob/master/other/python/cosmodistance.ipynb

Javascript Online Calculator:
https://mikehelland.github.io/hubbles-law/other/flrw.htm

The Python version shows CosmoloPy's results for comparison.

The Javascript version shows the results of Ned Wright's calculator for comparison.

----

I'm a skeptic of the expanding universe, as it seems many of you are. However, it's been useful to be able to calculate distances and times and things in the expanding universe, and learning a lot as I've gone. I came up with an algorithm that might be novel, not sure. Can't find other examples of it.

Here it is, I'd appreciate your feedback.

https://mikehelland.github.io/hubbles-law/papers/simple_flrw.pdf

**Comment by bgreenman** [comment]
Measuring the proton backword is a good way to calculate distance,but it will only show the photon's didtance,not how lare the universe is. We need a measuring method that could go far beyond current photon distance in order to find out what are our shortcomings in measuring cosmollogical physics

**Comment by mikehelland** [comment]
Here's a python version plotted against CosmoloPy:

https://github.com/mikehelland/hubbles-law/blob/master/other/python/cosmodistance.ipynb

**Reply by bgreenman** [reply]
good idea

**Comment by RedshiftDrift** [comment]
@mikehelland indeed the first thing to do when proposing a cosmological model is to show how to calculate angular and luminosity distances, and time elapsed, all as a function of redshift.  That way anybody can compare the model's prediction with observations.

[Many models](https://cosmology.info/redshift/) produce a logarithmic function of $1+z$, and many of those propose 'time variation' as a fundamental mechanism. (This is not novel.)  On this forum, @HanDeBruijn's model is similar - he'd be the best to comment.

Since you have the code, you could post an online calculator.  I had mine on my personal webpage but moved [it here](https://cosmology.info/code/marmet_l/cosmology-calculators.html) recently.  It calculates LCDM and STR side-by-side.  If you like you can copy mine [from the repository] (https://github.com/a-cosmology-group/acg/tree/main/code/marmet_l) and substitute your code to calculate the various quantities.

**Reply by mikehelland** [reply]
The model is standard FLRW/LCDM. 

I just thought the algorithm was particularly tidy. Instead of integrating over z to find comoving distance, this integrates over time to find proper distance and lookback time, and calculates z along the way. 

So it can find distance given redshift, or a redshift given a distance, or a redshift given a value for the Hubble parameter. It seemed pretty versatile that way.

**Reply by HanDeBruijn** [reply]
Thanks, Louis, for throwing the bait. But for me the thread named [_Time Dilated Past hypothesis_](https://github.com/orgs/a-cosmology-group/discussions/160) is a more appropriate place to discuss things.

**Comment by bgreenman** [comment]
RedshiftDrift calculation method to measure redshift of photons is one of the best available
